Chocotastic/CartViewController.swift

```diff
@@ -40,6 +44,34 @@ extension CartViewController {
     super.viewDidLoad()
     title = "Cart"
     configureFromCart()
+    setupCellConfiguration()
+    setupTotalCostConfiguration()
+  }
+}
+
+//MARK: - Rx Setup
+private extension CartViewController {
+  // ショッピングカート内のチョコレートをテーブルビューにバインド
+  func setupCellConfiguration() {
+    let cart = ShoppingCart.sharedCart
+    cart.chocolates
+      .map { Set($0) } // チョコレートの種類をユニーク化
+      .map { $0.sorted { $0.countryName > $1.countryName } } // セルの並びを国名でソート
+      .bind(to: tableView.rx.items(cellIdentifier: CartCell.Identifier, cellType: CartCell.self)) { index, chocolate, cell in
+        cell.configure(with: chocolate, cart: cart)
+      }
+      .disposed(by: disposeBag)
+  }
+  
+  // ショッピングカート内の合計値の変化を監視
+  func setupTotalCostConfiguration() {
+    let cart = ShoppingCart.sharedCart
+    cart.chocolates
+      .subscribe(onNext: { [unowned self] _ in
```

Comment:
非同期のタイミングによってselfがdeinitされている場合にクラッシュすることがまれにあるため、
クロージャからのself参照は、弱参照とするケースが多いです。

```suggestion
      .subscribe(onNext: { [weak self] _ in
```